Objects/odictobject.c

```diff
@@ -1045,30 +1045,28 @@ OrderedDict_setdefault_impl(PyODictObject *self, PyObject *key,
 
 /* pop() */
 
-PyDoc_STRVAR(odict_pop__doc__,
-"od.pop(k[,d]) -> v, remove specified key and return the corresponding\n\
-        value.  If key is not found, d is returned if given, otherwise KeyError\n\
-        is raised.\n\
-\n\
-        ");
-
 /* forward */
 static PyObject * _odict_popkey(PyObject *, PyObject *, PyObject *);
 
 /* Skips __missing__() calls. */
-static PyObject *
-odict_pop(PyObject *od, PyObject *args, PyObject *kwargs)
-{
-    static char *kwlist[] = {"key", "default", 0};
-    PyObject *key, *failobj = NULL;
+/*[clinic input]
+OrderedDict.pop
 
-    /* borrowed */
-    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|O:pop", kwlist,
-                                     &key, &failobj)) {
-        return NULL;
-    }
+    key: object
+    default: object = NULL
+
+Remove specified key and return the corresponding value.
+
+If key is not found, default is returned if given, otherwise KeyError
```

Comment:
We can make this read a little better, "If the key is not found, return the default if given;  otherwise, raise a KeyError."

Reply:
Thank you for your suggestion.